pennylane/_device.py

```diff
@@ -571,6 +573,56 @@ def my_expansion_function(self, tape, max_expansion=10):
         """
         self.custom_expand_fn = types.MethodType(fn, self)
 
+    def _local_tape_expand(self, tape, depth, stop_at):
```

Comment:
We cannot upgrade the existing expand method instead of creating a new method?

Reply:
The problem is that we want to selectively expand things because we have special logic for splitting hamiltonians. This bug only exists for the current default qubit, and the new defaultqubit2 device doesn't have this problem. I think it should be good enough to just add this fix in here until we swap over to the new device.

Reply:
This function should be removed when the old device and Hamiltonian are removed?

Reply:
I'm pushing for a future free from the baggage of `tape.expand`, and this helps us move in that direction.